placement/cluster_mapping_provider.go

```diff
@@ -106,21 +96,34 @@ func NewClusterMappingProvider(
 
 // QueryMappings returns the mappings for a given shard and retention policy
 func (mp *clusterMappingProvider) QueryMappings(
-	shard uint32, start, end time.Time) storage.ClusterMappingIter {
-
+	shard uint32, start, end time.Time) (storage.ClusterMappingIter, error) {
```

Comment:
Ah, did not realize we were planning on others implementing their own providers.  Thought others would simply provide a different `kv.Store` and not necessarily another `ClusterMappingProvider`.